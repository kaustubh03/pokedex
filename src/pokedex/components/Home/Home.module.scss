@import '../../commonstyles/variables.scss';

.container {
  display: flex;
  justify-content: center;
  align-items: center;
  .homeBanner{
    flex-direction: column;
    display: flex;
    justify-content: center;
    align-items: center;
  }
}

.containerPage{
  padding: 20px;
  background: rgba(1, 140, 239, 0.95);
  display: flex;
  justify-content:flex-start;
  align-items: flex-start;
  padding-left:16px;
  img{
    width:250px;
  }
}
