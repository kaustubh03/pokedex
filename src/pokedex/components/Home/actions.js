/*
  @Module : Pokedex
  @Component : Pokedex Search
  @Type : Action Creator
  @Description : Action for Pokedex Home.
  @Author : Kaustubh Saxena
*/

/*
  Action Type Constants
*/

import {
  FETCH_ALL_POKEMONS_LOADING,
  FETCH_ALL_POKEMONS_SUCCESS,
  FETCH_ALL_POKEMONS_FAILURE,
  FETCH_ALL_POKEMONS_CLEAR,
  FETCH_POKEMON_CARD_LOADING,
  FETCH_POKEMON_CARD_SUCCESS,
  FETCH_POKEMON_CARD_FAILURE,
  FETCH_POKEMON_CARD_CLEAR
} from '../../constants/actionTypes';

/*
  Fetch Wrapper used to hit API.
*/
import httpFetch from '../../utils/http';
/*
  Utility Imports
*/
import { fetchAllPokemons } from '../../utils/urlconfig';


let headers = {
  'Cache-Control': 'no-cache',
  'Accept':'application/json'
};

/*
  @Purpose : Get All Pokemons
  @System : PokeAPI
  @params : offset=number, limit=number

  P.S. - To optimise performance we can bring data in chunks and modify the value.
  Todo -- Optimise API Data to bring in chunks.
*/

export const getAllPokemonList = (offset = 0, limit = 1000) => {
    let url = `${fetchAllPokemons()}?offset=${offset}&limit=${limit}`;
  return {
    types: [
        FETCH_ALL_POKEMONS_LOADING,
        FETCH_ALL_POKEMONS_SUCCESS,
        FETCH_ALL_POKEMONS_FAILURE
    ],
    promise: () =>
        httpFetch(url, { headers })
  };
};
/*
  @Purpose : Get All Pokemons from Previously Saved List
  @System : Self
*/

export const rehydratePokemonList = () => {
    let localStorageObj =
      localStorage.getItem("pokemons") &&
      JSON.parse(localStorage.getItem("pokemons"));
    
    return {
      type: FETCH_ALL_POKEMONS_SUCCESS,
      result: localStorageObj
    };
};

/*
  @Purpose : Clear Fetched Pokemons from Redux Store
  @System : Self
*/
export function clearPokemonList(nextOption) {
  return {
    type: FETCH_ALL_POKEMONS_CLEAR,
    result: {}
  };
}

/*
  @Purpose : Get Selected Pokemon Details
  @System : PokeAPI
  @params : url=string
*/
export const getPokeCardDetail = (url) => {
    return {
      types: [
        FETCH_POKEMON_CARD_LOADING,
        FETCH_POKEMON_CARD_SUCCESS,
        FETCH_POKEMON_CARD_FAILURE
      ],
      promise: () => httpFetch(url, { headers })
    };
};


/*
  @Purpose : Clear Fetched Pokemons from Redux Store
  @System : Self
*/
export function clearPokemonCardDetail(nextOption) {
    return {
      type: FETCH_POKEMON_CARD_CLEAR,
      result: {}
    };
}
