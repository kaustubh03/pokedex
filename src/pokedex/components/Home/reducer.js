/*
  @Module : Pokedex
  @Component : Pokedex Search
  @Type : Reducer
  @Description : Reducer for Pokedex Home Screen
  @Author : Kaustubh Saxena
*/

/*
  Action Type Constants
*/

import {
  FETCH_ALL_POKEMONS_LOADING,
  FETCH_ALL_POKEMONS_SUCCESS,
  FETCH_ALL_POKEMONS_FAILURE,
  FETCH_ALL_POKEMONS_CLEAR,
  FETCH_POKEMON_CARD_LOADING,
  FETCH_POKEMON_CARD_SUCCESS,
  FETCH_POKEMON_CARD_FAILURE,
  FETCH_POKEMON_CARD_CLEAR
} from "../../constants/actionTypes";

// Utility Imports
import {
  getLoadingState,
  getSuccessState,
  getErrorState
} from "../../store/stateSettler";


const initialState = {
  pokemons: {
    loading: false,
    data: null,
    error: null
  },
  pokeCard: {
    loading: false,
    data: null,
    error: null
  }
};

export default function(state = initialState, action) {
  switch (action.type) {
    case FETCH_ALL_POKEMONS_LOADING: {
      return {
        ...state,
        pokemons: getLoadingState("pokemons", state)
      };
    }
    case FETCH_ALL_POKEMONS_SUCCESS: {
      return {
        ...state,
        pokemons: getSuccessState("pokemons", state, action)
      };
    }
    case FETCH_ALL_POKEMONS_FAILURE: {
      return {
        ...state,
        pokemons: getErrorState("pokemons", state, action)
      };
    }
    case FETCH_ALL_POKEMONS_CLEAR: {
      return {
        ...state,
        pokemons: initialState.pokemons
      };
    }
    case FETCH_POKEMON_CARD_LOADING: {
      return {
        ...state,
        pokeCard: getLoadingState("pokeCard", state)
      };
    }
    case FETCH_POKEMON_CARD_SUCCESS: {
      return {
        ...state,
        pokeCard: getSuccessState("pokeCard", state, action)
      };
    }
    case FETCH_POKEMON_CARD_FAILURE: {
      return {
        ...state,
        pokeCard: getErrorState("pokeCard", state, action)
      };
    }
    case FETCH_POKEMON_CARD_CLEAR: {
      return {
        ...state,
        pokeCard: initialState.pokeCard
      };
    }

    default:
      return state;
  }
}
