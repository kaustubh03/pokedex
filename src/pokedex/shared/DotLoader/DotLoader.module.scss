@import '../../commonstyles/variables.scss';

.wrapper {
  display: flex;
  justify-content:center;
  align-items: center;
  
  img{
    width:150px;
    height:150px;
  }
}
