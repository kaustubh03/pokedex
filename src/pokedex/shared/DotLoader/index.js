import React from 'react';
import s from './DotLoader.module.scss';
import pokeball from "../../public/images/pokeball.gif";


export default class DotLoader extends React.PureComponent {
  render() {
    return (
      <div className={s.wrapper}>
        <img src={pokeball} alt={'loader'}/>
      </div>
    );
  }
}
