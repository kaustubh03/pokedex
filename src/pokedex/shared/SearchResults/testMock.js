export const testData = [
    {
        "name": "clefable",
        "url": "https://pokeapi.co/api/v2/pokemon/36/"
    },
    {
        "name": "abra",
        "url": "https://pokeapi.co/api/v2/pokemon/63/"
    },
    {
        "name": "kadabra",
        "url": "https://pokeapi.co/api/v2/pokemon/64/"
    },
    {
        "name": "krabby",
        "url": "https://pokeapi.co/api/v2/pokemon/98/"
    },
    {
        "name": "electabuzz",
        "url": "https://pokeapi.co/api/v2/pokemon/125/"
    },
    {
        "name": "kabuto",
        "url": "https://pokeapi.co/api/v2/pokemon/140/"
    },
    {
        "name": "kabutops",
        "url": "https://pokeapi.co/api/v2/pokemon/141/"
    },
    {
        "name": "sableye",
        "url": "https://pokeapi.co/api/v2/pokemon/302/"
    },
    {
        "name": "swablu",
        "url": "https://pokeapi.co/api/v2/pokemon/333/"
    },
    {
        "name": "absol",
        "url": "https://pokeapi.co/api/v2/pokemon/359/"
    },
    {
        "name": "gabite",
        "url": "https://pokeapi.co/api/v2/pokemon/444/"
    },
    {
        "name": "abomasnow",
        "url": "https://pokeapi.co/api/v2/pokemon/460/"
    },
    {
        "name": "karrablast",
        "url": "https://pokeapi.co/api/v2/pokemon/588/"
    },
    {
        "name": "vullaby",
        "url": "https://pokeapi.co/api/v2/pokemon/629/"
    },
    {
        "name": "flabebe",
        "url": "https://pokeapi.co/api/v2/pokemon/669/"
    },
    {
        "name": "pumpkaboo-average",
        "url": "https://pokeapi.co/api/v2/pokemon/710/"
    },
    {
        "name": "charjabug",
        "url": "https://pokeapi.co/api/v2/pokemon/737/"
    },
    {
        "name": "crabrawler",
        "url": "https://pokeapi.co/api/v2/pokemon/739/"
    },
    {
        "name": "crabominable",
        "url": "https://pokeapi.co/api/v2/pokemon/740/"
    },
    {
        "name": "pumpkaboo-small",
        "url": "https://pokeapi.co/api/v2/pokemon/10027/"
    },
    {
        "name": "pumpkaboo-large",
        "url": "https://pokeapi.co/api/v2/pokemon/10028/"
    },
    {
        "name": "pumpkaboo-super",
        "url": "https://pokeapi.co/api/v2/pokemon/10029/"
    },
    {
        "name": "absol-mega",
        "url": "https://pokeapi.co/api/v2/pokemon/10057/"
    },
    {
        "name": "abomasnow-mega",
        "url": "https://pokeapi.co/api/v2/pokemon/10060/"
    },
    {
        "name": "sableye-mega",
        "url": "https://pokeapi.co/api/v2/pokemon/10066/"
    }
];