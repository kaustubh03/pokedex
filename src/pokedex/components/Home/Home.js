/*
  @Module : Pokedex
  @Component : Home
  @Type : Screen
  @Description : Entry Screen
  @Author : Kaustubh Saxena 
*/

// React Imports
import React from 'react';

// Component Imports
import DotLoader from "../../shared/DotLoader";
import Search from "../../shared/Search/Search";
import SearchResults from "../../shared/SearchResults/SearchResults";
import PokeCard from "../../shared/PokeCard/PokeCard";

// Stylesheet Imports
import s from './Home.module.scss';

// Asset and Utility Imports
import banner from "../../public/images/banner.png";
import { performSearch, timeDifference } from "../../utils/helper";


export default class Home extends React.Component {
  // State Declaration
  state = {
    showSearch:false,
    searchResults:null,
    showDetails:false
  };

  // Lifecycle Methods
  componentDidMount() {
    const { getAllPokemonList, rehydratePokemonList } = this.props;
    let savedPokemonList = localStorage.getItem('pokemons');
    let validity = null;
    if(savedPokemonList){
      let _date = JSON.parse(savedPokemonList).date;
      validity = timeDifference(_date);
    }
    // Fetch All Pokemon List Before Search
    if (!localStorage.getItem("pokemons") || validity) {
      // Check Validity and Remove the Stale Data
      if(validity){
        localStorage.removeItem("pokemons");
      }
      getAllPokemonList();
    } else {
      rehydratePokemonList();
    }
  }

  componentDidUpdate = (prevProps, prevState) =>{
    const { pokemons, pokeCard } = this.props;
      if (prevProps.pokemons.loading && !pokemons.loading && pokemons.data) {
        this.setState({ showSearch: true });
        // Append A Date To Result Object and save to Local Storage
        let _pokemons = { ...pokemons };
        _pokemons.data['date'] = new Date();
        localStorage.setItem('pokemons', JSON.stringify(_pokemons.data));
        this.getRandomCardDetail();
      }
      /*
        For Rehydrating Case
      */
      if(!prevProps.pokemons.loading && !pokemons.loading && !prevProps.pokemons.data && pokemons.data && localStorage.getItem('pokemons')){
        this.setState({ showSearch: true });
        this.getRandomCardDetail();
      }
    if (
      prevProps.pokeCard.loading &&
      !pokeCard.loading &&
      pokeCard.data
    ) {
      let _pokeCard = pokeCard.data;
      _pokeCard['date'] = new Date();
      localStorage.setItem("pokeCardData", JSON.stringify(_pokeCard));
    }
  }

  getRandomCardDetail = () =>{
    const { pokemons, getPokeCardDetail } = this.props;
    let count = pokemons.data && pokemons.data.count && pokemons.data.count;
    if(count){
      let random = Math.floor(Math.random() * count) + 1;
      let randomPokemonObj = pokemons.data.results[random];

      let pokeCardValidity = null;
      let savedPokeCardData = localStorage.getItem("pokeCardData");
      if (savedPokeCardData) {
        let _date = JSON.parse(savedPokeCardData).date;
        pokeCardValidity = timeDifference(_date);
      }
      if (!localStorage.getItem("pokeCardData") || pokeCardValidity) {
        getPokeCardDetail(randomPokemonObj.url);
      }
    }
  }

  /*
    Get Body of Screen on the basis of Action.
    Possible cases are Pokecard or Search Results
    @params : none
  */
  getBody = () =>{
    const { pokeCard } = this.props;
    const { searchResults, showDetails } = this.state; 
    let pokeCardData = localStorage.getItem("pokeCardData") ? JSON.parse(localStorage.getItem("pokeCardData")):pokeCard.data 
    if (!searchResults){
      return (
        <div>
          {((!pokeCard.loading && pokeCard.data) ||
            localStorage.getItem("pokeCardData")) && (
            <PokeCard data={pokeCardData} />
          )}
          {pokeCard.loading && <DotLoader />}
        </div>
      );
    }
    else{
      return (
        <div>
          {!showDetails && <SearchResults
            results={searchResults}
            handleResultClick={this.handleResultClick}
          />}
        </div>
      );
    } 
  }

  /*
    Handle Search Result Click Callback
    @params : searchItem = Object
  */
  handleResultClick = (searchItem) =>{
    const { history } = this.props;
    if(searchItem && searchItem.url){
      let url = btoa(searchItem.url);
      history.push(`/detail/${url}`);
    }
  }

  /*
    Handle Change Callback for Search Field
    @params : event = Object
  */
  handleSearch = (e) =>{
    const { pokemons } = this.props;
    let value = e.target.value;
    /*
      If a valid Value is there. It will perform a search if a non empty value is present.
      If a Null Value is given. It will revert back to Pokecard Screen.
    */
    if(value && value!==""){
      let searchResults = performSearch(pokemons.data, value);
      this.setState({ searchResults: searchResults });
    }
    else{
      this.setState({searchResults:null})
    }

  }

  render() {
    const { showSearch } = this.state;
    let parentClassName = showSearch ? 'containerPage':'container';
    return (
      <div>
        <div className={s[parentClassName]}>
            <div className={s.homeBanner}>
              <img src={banner} alt={'banner'}/>
            {!showSearch &&
              <>
                <DotLoader />
              </>
            }
            </div>
            {showSearch && <Search searchCallback={this.handleSearch} />}
        </div>
        {showSearch && this.getBody()}
      </div>
      
    );
  }
}
