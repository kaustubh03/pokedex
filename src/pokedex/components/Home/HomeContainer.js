
// Redux Imports
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  getAllPokemonList,
  rehydratePokemonList,
  getPokeCardDetail
} from "./actions";

// React Imports
import { withRouter } from 'react-router-dom';

// Class To Contain
import Home from './Home';


const mapStateToProps = (state, prevProps) => {
  return {
    pokemons: state.homeReducer.pokemons,
    pokeCard: state.homeReducer.pokeCard
  };
};

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      getAllPokemonList,
      rehydratePokemonList,
      getPokeCardDetail
    },
    dispatch
  );
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Home)
);
