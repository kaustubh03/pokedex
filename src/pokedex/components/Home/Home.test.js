import React from "react";
import Home from "./Home";
import { loadingData, successData } from "./testMock.js";
import { mount, configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import DotLoader from "../../shared/DotLoader";

configure({ adapter: new Adapter() });

test("If API is Loading. Show Loader", () => {
  const wrapper = mount(
    <Home getAllPokemonList={() => {}} pokemons={loadingData} />
  );
  expect(wrapper.find(DotLoader).length).toBe(1);
});

describe("Home", () => {
  it('should render correctly', () => {
    const component = shallow(
      <Home getAllPokemonList={() => {}} pokemons={successData} />
    );
    expect(component).toMatchSnapshot();
  });
});