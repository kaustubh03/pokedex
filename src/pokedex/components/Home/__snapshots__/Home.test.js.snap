// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    getAllPokemonList={[Function]}
    pokemons={
      Object {
        "data": Object {
          "count": 20,
          "next": "https://pokeapi.co/api/v2/pokemon/?offset=40&limit=20",
          "previous": "https://pokeapi.co/api/v2/pokemon/?offset=0&limit=20",
          "results": Array [
            Object {
              "name": "spearow",
              "url": "https://pokeapi.co/api/v2/pokemon/21/",
            },
            Object {
              "name": "fearow",
              "url": "https://pokeapi.co/api/v2/pokemon/22/",
            },
            Object {
              "name": "ekans",
              "url": "https://pokeapi.co/api/v2/pokemon/23/",
            },
            Object {
              "name": "arbok",
              "url": "https://pokeapi.co/api/v2/pokemon/24/",
            },
            Object {
              "name": "pikachu",
              "url": "https://pokeapi.co/api/v2/pokemon/25/",
            },
            Object {
              "name": "raichu",
              "url": "https://pokeapi.co/api/v2/pokemon/26/",
            },
            Object {
              "name": "sandshrew",
              "url": "https://pokeapi.co/api/v2/pokemon/27/",
            },
            Object {
              "name": "sandslash",
              "url": "https://pokeapi.co/api/v2/pokemon/28/",
            },
            Object {
              "name": "nidoran-f",
              "url": "https://pokeapi.co/api/v2/pokemon/29/",
            },
            Object {
              "name": "nidorina",
              "url": "https://pokeapi.co/api/v2/pokemon/30/",
            },
            Object {
              "name": "nidoqueen",
              "url": "https://pokeapi.co/api/v2/pokemon/31/",
            },
            Object {
              "name": "nidoran-m",
              "url": "https://pokeapi.co/api/v2/pokemon/32/",
            },
            Object {
              "name": "nidorino",
              "url": "https://pokeapi.co/api/v2/pokemon/33/",
            },
            Object {
              "name": "nidoking",
              "url": "https://pokeapi.co/api/v2/pokemon/34/",
            },
            Object {
              "name": "clefairy",
              "url": "https://pokeapi.co/api/v2/pokemon/35/",
            },
            Object {
              "name": "clefable",
              "url": "https://pokeapi.co/api/v2/pokemon/36/",
            },
            Object {
              "name": "vulpix",
              "url": "https://pokeapi.co/api/v2/pokemon/37/",
            },
            Object {
              "name": "ninetales",
              "url": "https://pokeapi.co/api/v2/pokemon/38/",
            },
            Object {
              "name": "jigglypuff",
              "url": "https://pokeapi.co/api/v2/pokemon/39/",
            },
            Object {
              "name": "wigglytuff",
              "url": "https://pokeapi.co/api/v2/pokemon/40/",
            },
          ],
        },
        "error": null,
        "loading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="container"
        >
          <div
            className="homeBanner"
          >
            <img
              alt="banner"
              src="banner.png"
            />
            <React.Fragment>
              <DotLoader />
            </React.Fragment>
          </div>
        </div>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="homeBanner"
            >
              <img
                alt="banner"
                src="banner.png"
              />
              <React.Fragment>
                <DotLoader />
              </React.Fragment>
            </div>,
            false,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="banner"
                  src="banner.png"
                />,
                <React.Fragment>
                  <DotLoader />
                </React.Fragment>,
              ],
              "className": "homeBanner",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "banner",
                  "src": "banner.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <DotLoader />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": Symbol(react.fragment),
              },
            ],
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="container"
          >
            <div
              className="homeBanner"
            >
              <img
                alt="banner"
                src="banner.png"
              />
              <React.Fragment>
                <DotLoader />
              </React.Fragment>
            </div>
          </div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="homeBanner"
              >
                <img
                  alt="banner"
                  src="banner.png"
                />
                <React.Fragment>
                  <DotLoader />
                </React.Fragment>
              </div>,
              false,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="banner"
                    src="banner.png"
                  />,
                  <React.Fragment>
                    <DotLoader />
                  </React.Fragment>,
                ],
                "className": "homeBanner",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "banner",
                    "src": "banner.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <DotLoader />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": Symbol(react.fragment),
                },
              ],
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Home should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    getAllPokemonList={[Function]}
    pokemons={
      Object {
        "data": Object {
          "count": 20,
          "next": "https://pokeapi.co/api/v2/pokemon/?offset=40&limit=20",
          "previous": "https://pokeapi.co/api/v2/pokemon/?offset=0&limit=20",
          "results": Array [
            Object {
              "name": "spearow",
              "url": "https://pokeapi.co/api/v2/pokemon/21/",
            },
            Object {
              "name": "fearow",
              "url": "https://pokeapi.co/api/v2/pokemon/22/",
            },
            Object {
              "name": "ekans",
              "url": "https://pokeapi.co/api/v2/pokemon/23/",
            },
            Object {
              "name": "arbok",
              "url": "https://pokeapi.co/api/v2/pokemon/24/",
            },
            Object {
              "name": "pikachu",
              "url": "https://pokeapi.co/api/v2/pokemon/25/",
            },
            Object {
              "name": "raichu",
              "url": "https://pokeapi.co/api/v2/pokemon/26/",
            },
            Object {
              "name": "sandshrew",
              "url": "https://pokeapi.co/api/v2/pokemon/27/",
            },
            Object {
              "name": "sandslash",
              "url": "https://pokeapi.co/api/v2/pokemon/28/",
            },
            Object {
              "name": "nidoran-f",
              "url": "https://pokeapi.co/api/v2/pokemon/29/",
            },
            Object {
              "name": "nidorina",
              "url": "https://pokeapi.co/api/v2/pokemon/30/",
            },
            Object {
              "name": "nidoqueen",
              "url": "https://pokeapi.co/api/v2/pokemon/31/",
            },
            Object {
              "name": "nidoran-m",
              "url": "https://pokeapi.co/api/v2/pokemon/32/",
            },
            Object {
              "name": "nidorino",
              "url": "https://pokeapi.co/api/v2/pokemon/33/",
            },
            Object {
              "name": "nidoking",
              "url": "https://pokeapi.co/api/v2/pokemon/34/",
            },
            Object {
              "name": "clefairy",
              "url": "https://pokeapi.co/api/v2/pokemon/35/",
            },
            Object {
              "name": "clefable",
              "url": "https://pokeapi.co/api/v2/pokemon/36/",
            },
            Object {
              "name": "vulpix",
              "url": "https://pokeapi.co/api/v2/pokemon/37/",
            },
            Object {
              "name": "ninetales",
              "url": "https://pokeapi.co/api/v2/pokemon/38/",
            },
            Object {
              "name": "jigglypuff",
              "url": "https://pokeapi.co/api/v2/pokemon/39/",
            },
            Object {
              "name": "wigglytuff",
              "url": "https://pokeapi.co/api/v2/pokemon/40/",
            },
          ],
        },
        "error": null,
        "loading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="container"
        >
          <div
            className="homeBanner"
          >
            <img
              alt="banner"
              src="banner.png"
            />
            <React.Fragment>
              <DotLoader />
            </React.Fragment>
          </div>
        </div>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="homeBanner"
            >
              <img
                alt="banner"
                src="banner.png"
              />
              <React.Fragment>
                <DotLoader />
              </React.Fragment>
            </div>,
            false,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="banner"
                  src="banner.png"
                />,
                <React.Fragment>
                  <DotLoader />
                </React.Fragment>,
              ],
              "className": "homeBanner",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "banner",
                  "src": "banner.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <DotLoader />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": Symbol(react.fragment),
              },
            ],
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="container"
          >
            <div
              className="homeBanner"
            >
              <img
                alt="banner"
                src="banner.png"
              />
              <React.Fragment>
                <DotLoader />
              </React.Fragment>
            </div>
          </div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="homeBanner"
              >
                <img
                  alt="banner"
                  src="banner.png"
                />
                <React.Fragment>
                  <DotLoader />
                </React.Fragment>
              </div>,
              false,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="banner"
                    src="banner.png"
                  />,
                  <React.Fragment>
                    <DotLoader />
                  </React.Fragment>,
                ],
                "className": "homeBanner",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "banner",
                    "src": "banner.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <DotLoader />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": Symbol(react.fragment),
                },
              ],
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
